pkg/storage/utils_test.go

```diff
@@ -0,0 +1,192 @@
+			}
+		})
+	}
+}
+
+func TestGetMachineReadableFormatForList(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		inputStorage []Storage
+		want         StorageList
+	}{
+		{
+			name: "test case 1: with a single pvc",
+			inputStorage: []Storage{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-1",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.dev/v1alpha1",
+					},
+					Spec: StorageSpec{
+						Size: "100Mi",
+						Path: "data",
+					},
+				},
+			},
+			want: StorageList{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "odo.dev/v1alpha1",
+				},
+				ListMeta: metav1.ListMeta{},
+				Items: []Storage{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "example-pvc-1",
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "List",
+							APIVersion: "odo.dev/v1alpha1",
+						},
+						Spec: StorageSpec{
+							Size: "100Mi",
+							Path: "data",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "test case 2: with multiple pvcs",
+			inputStorage: []Storage{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-0",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.dev/v1alpha1",
+					},
+					Spec: StorageSpec{
+						Size: "100Mi",
+						Path: "data",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-1",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.dev/v1alpha1",
+					},
+					Spec: StorageSpec{
+						Size: "500Mi",
+						Path: "backend",
+					},
+				},
+			},
+			want: StorageList{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "odo.dev/v1alpha1",
+				},
+				ListMeta: metav1.ListMeta{},
+				Items: []Storage{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "example-pvc-0",
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "List",
+							APIVersion: "odo.dev/v1alpha1",
+						},
+						Spec: StorageSpec{
+							Size: "100Mi",
+							Path: "data",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "example-pvc-1",
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "List",
+							APIVersion: "odo.dev/v1alpha1",
+						},
+						Spec: StorageSpec{
+							Size: "500Mi",
+							Path: "backend",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotStorage := GetMachineReadableFormatForList(tt.inputStorage)
+			if !reflect.DeepEqual(tt.want, gotStorage) {
+				t.Errorf("the returned storage is different, expected: %v, got: %v", tt.want, gotStorage)
+			}
+		})
+	}
+}
+
+func generateStorage(storage Storage, status StorageStatus, containerName string) Storage {
+	storage.Status = status
+	storage.Spec.ContainerName = containerName
+	return storage
+}
```

Comment:
It seems like this function is not in use. Can you remove it?

Reply:
It is currently being used by the kubernetes client tests.

Reply:
My bad, Idk how I didn't notice that before.

Reply:
In future, other client implementations can use this function thus we should keep it here.